src/Services/OpenApiGenerator.php

```diff
@@ -366,4 +315,243 @@ private function getResponses(Route $route): array
+            }
+
+            return $cache[$ruleSpec] = $parsed;
+        }
+
+        return $cache[$ruleSpec];
+    }
+
+    private function normalizeRules(array $rawRules): array
+    {
+        $rules = [];
+        foreach ($rawRules as $propertyPath => $ruleSpec) {
+            $this->setDeep($rules, $propertyPath, $this->normalizeRuleSpec($ruleSpec));
+        }
+
+        return $rules;
+    }
+
+    private function normalizeRuleSpec(string|array $ruleSpec): string
+    {
+        if (is_string($ruleSpec)) {
+            return $ruleSpec;
+        }
+        $specs = [];
+        foreach ($ruleSpec as $key => $val) {
+            if (is_object($val)) {
+                $val = '';
+            }
+            if (is_numeric($key)) {
+                $key = $val;
+                $val = '';
+            } elseif (is_array($val)) {
+                $val = implode(',', $val);
+            }
+            if ($key) {
+                $specs[] = $key . ($val ? ':' : '') . $val;
+            }
+        }
+
+        return implode('|', $specs);
+    }
+
+    private function setDeep(&$data, $path, $value)
+    {
+        $keys    = explode('.', $path);
+        $lastKey = array_pop($keys);
+        $temp    = &$data;
+
+        while ($keys) {
+            $key = array_shift($keys);
+            $temp[$key] ??= [];
+            if (is_string($temp[$key])) {
+                $temp[$key] = [
+                    '#rule' => $temp[$key],
+                ];
+            }
+            $temp = &$temp[$key];
+        }
+
+        $temp[$lastKey] = $value;
+    }
+
+    private function buildDefaultRuleHandlers()
+    {
+        // Map validation rule to schema `type`.
+        $ruleTypeMap = [
+            'boolean' => 'boolean',
+            'numeric' => 'number',
+            'array'   => 'array',
+            'integer' => 'integer',
+        ];
+        $ruleNames = implode('|', array_keys($ruleTypeMap));
+        $this->ruleHandlers["/^($ruleNames)$/"] = function (
+            array $schema,
+            string $rule,
+        ) use ($ruleTypeMap) {
+            return array_merge($schema, ['type' => $ruleTypeMap[$rule]]);
+        };
+
+        // Map validation rule to schema `format`.
+        $ruleFormatMap = [
+            'uuid'          => 'uuid',
+            'integer'       => 'int64',
+            'email'         => 'email',
+            'ipv4'          => 'ipv4',
+            'ipv6'          => 'ipv6',
+            'ip'            => 'ip',
+            'url'           => 'url',
+            'uploaded_file' => 'binary',
+        ];
+        $ruleNames = implode('|', array_keys($ruleFormatMap));
+        $this->ruleHandlers["/^($ruleNames)$/"] = function (
+            array $schema,
+            string $rule,
+        ) use ($ruleFormatMap) {
+            return array_merge($schema, ['format' => $ruleFormatMap[$rule]]);
+        };
+
+        $this->ruleHandlers['/date/'] = function (array $schema) {
```

Comment:
The default "date" validation rule in `rakit/validation` only validates `Y-m-d` which would be the `date` type. `date-time` is for a full date of the form: `Y-m-dTH:i:sZ`. Might be an idea to either add an additional `datetime` rule (that would need implementing in the form request bundle) or maybe a regex on the format string; though I think this would be far too complicated for the sake of more accurate docs.